src/graphql/resolvers.ts

```diff
@@ -1,28 +1,41 @@
 import { login } from 'datasource/login';
 import { createUser } from 'datasource/create-user';
-import { CreateuserInterface, LoginInterface } from 'graphql/interfaces';
-import { tokenIsValid } from 'provider/token-validation-provider';
+import { CreateuserInterface, IDInterface, LoginInterface } from 'graphql/interfaces';
+import { isTokenValid } from 'provider/token-validation-provider';
 import { CustomError } from 'error/errors';
+import { queryUser } from 'datasource/query-user';
 
 export const resolvers = {
   Query: {
     hello: (): String => 'Hello, world!',
+    user: async (_: unknown, { id }: IDInterface, context) => {
+      if (!isTokenValid(context['token'])) {
+        throw new CustomError('JWT inválido.', 401, 'Operação não autorizada.');
+      }
+      const { user } = await queryUser(id);
+      return { user };
+    },
```

Comment:
Dá pra dar uma simplificada na função:

```suggestion
    user: (_, { id }: IDInterface, context) => {
      if (!isTokenValid(context['token'])) {
        throw new CustomError('JWT inválido.', 401, 'Operação não autorizada.');
      }

      return queryUser(id);
    },
```

1. A variável nomeada com `_` já indica que não será usada, então acho que nem precisa tipar
2. O async/await não precisa porque o `queryUser` já retorna uma Promise de objeto `{ user: UserX }`
3. o `const { user }` seguido de `return { user }` pode ser simplificado pra retornar direto o valor que a função retorna

Reply:
Mas se o queryUser retorna uma Promise, eu não devia retornar isso no resolver, pq ainda vai estar pendente, não? É o que parece estar acontecendo agora que segui essa sugestão

Reply:
Nevermind, não era isso! Tá bem check o que vc falou 😅